gobblin-data-management/src/main/java/gobblin/data/management/copy/WatermarkBasedCopyableFileFilter.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.hadoop.fs.FileSystem;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Preconditions;
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+import gobblin.configuration.SourceState;
+import gobblin.configuration.WorkUnitState;
+import gobblin.data.management.copy.watermark.CopyEntityWatermarkGenerator;
+import gobblin.data.management.copy.watermark.CopyEntityWatermarkHelper;
+import gobblin.source.extractor.ComparableWatermark;
+import gobblin.source.extractor.Watermark;
+import gobblin.source.extractor.WatermarkInterval;
+import gobblin.source.extractor.WatermarkSerializerHelper;
+
+
+/**
+ * Implementation of {@link CopyableFileFilter} that filters {@link CopyableFile}s based on previous watermark information.
+ */
+@Slf4j
+public class WatermarkBasedCopyableFileFilter implements CopyableFileFilter {
+  /**
+   * Only create work units that have higher expected high watermark than the previous actual watermark.
+   */
+  public static final String FILTER_BY_WATERMARK = CopyConfiguration.COPY_PREFIX + ".filterWorkUnitsByWatermark";
```

Comment:
Why is this needed? Is this filter always instantiated or is it user specified? If the latter, wouldn't the user specifying this filter imply that they want the filtering?